File: dotnet/Endpoints/GetAll.cs
using Microsoft.AspNetCore.Http.HttpResults;
using Microsoft.AspNetCore.Mvc;
using Todo.Core.Abstract.Services;
using Todo.Endpoints.Mappers;
using Todo.Endpoints.Models;

namespace Todo.Endpoints;

internal class GetAll
{
    public static async Task<Ok<ICollection<TodoDto>>>
        Execute([FromServices] ITodoService todoService)
    {
        var todos = await todoService.GetAll();

        return TypedResults.Ok(todos.ToDtos());
    }
}

File: nodejs/src/endpoints/todos/get-all.ts
import { ITodoService } from "../../core/abstract/services/todo-service";
import { Request, Response } from "express";

export const GetAllTodosEndpoint = (todoService: ITodoService) => {
  return async (req: Request, res: Response) => {
    const todos = await todoService.getAll();

    res.json(todos);
  };
};
